### ISSUE TITLE
Secondary axis doesn't show axis ticks or labels with coord_trans

### ISSUE THREAD

**Issue by anthonytw** [tidyverse/ggplot2#2990]
Exactly the same issue as #2072, which I take it was fixed since it is closed.

The secondary axis ticks and labels do not show up when coord_trans is present, even if using the identity transformation. The axis label ("name" parameter) does still appear.

For example, roughly taking from the examples in sec_axis:

```r
ggplot(mtcars, aes(cyl, mpg)) + geom_point()  +
    scale_y_continuous(sec.axis = sec_axis(~.+10, name='MPG + 10'))
```

works as expected, but

```r
ggplot(mtcars, aes(cyl, mpg)) + geom_point()  +
    scale_y_continuous(sec.axis = sec_axis(~.+10, name='MPG + 10')) +
    coord_trans()
```

loses the axis ticks and labels (but keeps the title "MPG + 10").

**Comment by dpseidel** [comment]
It seems so. At least it predates my changes from #2796 and is present with version 3.0.0.

**Comment by ptoche** [comment]
The axis is dropped altogether, not just the labels and ticks, as can be seen by adding a theme like ``theme_classic()`` to the previous examples. This is the third case of a missing axis I see in one week. Was there a change made to how null breaks/labels are handled that might explain why axes are being dropped in different contexts?

**Comment by yutannihilation** [comment]
This problem comes from here in `CoordTrans$setup_panel_params()`; `out` is redefined so the params for the secondary axis (`out$sec.range`, `out$sec.major`, `out$sec.minor`, etc) are lost.

https://github.com/tidyverse/ggplot2/blob/868fdb7c9176be685850a3a698af7acb5e890a4a/R/coord-transform.r#L192-L195

If I replace the code with below, at least the labels are shown (ticks are wrong, though).

``` r
  out$major <- out$major_source
  out$minor <- out$minor_source
  out$arrange <- scale$axis_order()
```

**Comment by dpseidel** [comment]
I played around with this a little on my branch containing my most recent sec.axis patch (#3040) and can confirm the misalignment is still an issue. Interestingly, it's unique to all the other misalignment issues. It seems that when `coord_trans()` is specified `sec.axis` is passed the unexpanded range -- the transformed range of the data, not the transformed range of the primary scale (as it is when no coord is specified). This leads to misalignment relative to the primary axis.
 
``` r
# loading local dpseidel/revert2796 + suggested coord_trans patch
library(ggplot2)

c <- ggplot(mtcars, aes(cyl, mpg)) +
  geom_point() +
  scale_y_continuous(sec.axis = sec_axis(~ . + 10, name = "MPG + 10")) +
  coord_trans()

```

A screenshot of output from within `debug(get("break_info", AxisSecondary))` to demonstrate the actual break specifications calculated: 

<img width="733" alt="screen shot 2018-12-16 at 10 44 58 am" src="https://user-images.githubusercontent.com/16724456/50057660-5fd7ba00-0122-11e9-9fd1-ff33dd2007f1.png">

Notably, and likely a separate bug worth flagging in its own issue,  the `layer_scales` function runs the `break_info` function with the expected, expanded, range even when passed `coord_trans()` and returns properly aligned break values; a bug I have encountered when debugging other sec.axis issues and which can give users a false reading of the position of tick axes (as it does below): 
 
``` r
layer_scales(c)$y$break_info()
#> $range
#> [1] 10.4 33.9
#> 
#> $labels
#> [1] "15" "20" "25" "30"
#> 
#> $major
#> [1] 0.1957447 0.4085106 0.6212766 0.8340426
#> 
#> $minor
#> [1] 0.0893617 0.1957447 0.3021277 0.4085106 0.5148936 0.6212766 0.7276596
#> [8] 0.8340426 0.9404255
#> 
#> $major_source
#> [1] 15 20 25 30
#> 
#> $minor_source
#> [1] 12.5 15.0 17.5 20.0 22.5 25.0 27.5 30.0 32.5
#> 
#> $sec.range
#> [1] 20.4 43.9
#> 
#> $sec.labels
#> [1] "25" "30" "35" "40"
#> 
#> $sec.major
#> [1] 0.1957447 0.4085106 0.6212766 0.8340426
#> 
#> $sec.minor
#> [1] 0.0893617 0.1957447 0.3021277 0.4085106 0.5148936 0.6212766 0.7276596
#> [8] 0.8340426 0.9404255
#> 
#> $sec.major_source
#> [1] 25 30 35 40
#> 
#> $sec.minor_source
#> [1] 22.5 25.0 27.5 30.0 32.5 35.0 37.5 40.0 42.5
```

<sup>Created on 2018-12-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Since this is a scale expansion issue, misalignment is fixed if one specifies `expand = c(0,0)` but we will likely want to dig deeper into this (and merge an appropriate coord_trans patch) once the sec.axis refactoring is complete.

**Comment by thomasp85** [comment]
@dpseidel do you have any idea how to handle this most gracefully?

**Comment by dpseidel** [comment]
I'll need to look back into it. I'll plan my own private dev day this weekend to dig into this and other open axis/scale issues :+1:

**Comment by paleolimbot** [comment]
This seems related to #3338 in that `coord_trans()` is handling scale expansion differently than other coords.
